### ISSUE TITLE
Add a tag to stop user from editing anything.

### ISSUE THREAD

**Issue by littlewhitecloud** [littlewhitecloud/TkTerminal#27]
I was testing the widget and I found we can edit the output and the path.
Then the widget became what a mess.
I through a simple solution to fix it:
- Create a uneditable tag.
- Add it to the path output and command output.

**Comment by Moosems** [comment]
I had fixed that with the left right up commands...

**Comment by littlewhitecloud** [comment]
？

**Comment by littlewhitecloud** [comment]
But I can still delete things.

**Comment by littlewhitecloud** [comment]
![image](https://github.com/littlewhitecloud/TkTerminal/assets/71159641/dcdb83b2-3704-4c37-9daa-8f0d4bc097f9)
![image](https://github.com/littlewhitecloud/TkTerminal/assets/71159641/1cdf23ef-7f05-482a-87a6-f9fee47c36d7)

**Comment by Moosems** [comment]
So clicking needs a bind.

**Comment by littlewhitecloud** [comment]
Is there a way to give a readonly tag?

**Comment by littlewhitecloud** [comment]
> So clicking needs a bind.

I prefer use a tag to set the text above it read only.

**Comment by Moosems** [comment]
There's no built in way but I can create one.

**Comment by littlewhitecloud** [comment]
> There's no built in way but I can create one.

WOW! How?

**Comment by Moosems** [comment]
Bind all keypresses and button clicks and determine if the end result would be on top of any of the tag ranges. If it is, cancel the action otherwise allow it. Up and down get special treatment.

**Comment by littlewhitecloud** [comment]
> Bind all keypresses and button clicks and determine if the end result would be on top of any of the tag ranges. If it is, cancel the action otherwise allow it. Up and down get special treatment.

Good idea, but bind a lot maybe make the program slow.

**Comment by Moosems** [comment]
It'll be fast enough :).

**Comment by Moosems** [comment]
@littlewhitecloud I'm going to do a ton of work tomorrow on the repo (eastern time US) so buckle up :D.

**Comment by littlewhitecloud** [comment]
> @littlewhitecloud I'm going to do a ton of work tomorrow on the repo (eastern time US) so buckle up :D.

So I think We should split it out from the improve pr, create a new one and starting our trip :)

**Comment by littlewhitecloud** [comment]
```python
self.index = 0
…
self.text.bind(“Left”, self._left)
self.text.bind(“Right”, self._right)
self.text.bind(“Click”, self._click)

def _left(self, event):
    self.index += 1
    if self.index > 0:
       self.text[“state”] = “normal”
def _right(self, event):
    self.index -= 1
    if self.index < 0:
        self.text[“state”] = “readonly”

def _click(self, event):
    self.index = #cursor index




```

**Comment by littlewhitecloud** [comment]
My form, but still have a lot of logic problem, fix it later.

**Comment by Moosems** [comment]
Clear also needs to be a special command for typing.

**Comment by Moosems** [comment]
Mine should be able to be put in a duct allowing for better handling. I'll need to finish the handler but I have a feeling it will work well.

**Comment by littlewhitecloud** [comment]
> Mine should be able to be put in a duct allowing for better handling. I'll need to finish the handler but I have a feeling it will work well.

How long will it take?

**Comment by littlewhitecloud** [comment]
If it still take a long time, maybe we should fix it in another pr. Just use mine version as a temp solution. When you are finished, we can remove mine can replace with your solution.

**Comment by Moosems** [comment]
Yeah. I'd merge the PR and when I have a free week I'll make a big update PR.
